src/sagemaker/config/config_schema.py

```diff
@@ -568,6 +584,55 @@ def _simple_path(*args: str):
             "minItems": 1,
             "maxItems": 15,
         },
+        # Regex is taken from https://docs.aws.amazon.com/sagemaker/latest/APIReference/API_DebugHookConfig.html
+        "collectionName": {
+            TYPE: "string",
+            "pattern": r".*",
+            "minLength": 1,
+            "maxLength": 256,
+        },
+        "collectionParameters": {
+            TYPE: OBJECT,
+            ADDITIONAL_PROPERTIES: False,
+            PATTERN_PROPERTIES: {
+                r"(.*){1,256}": {
+                    TYPE: "string",
+                    "pattern": r".*",
+                    "maxLength": 256,
+                }
+            },
+            "maxProperties": 20,
+        },
+        "collectionConfiguration": {
+            TYPE: OBJECT,
+            ADDITIONAL_PROPERTIES: False,
+            PROPERTIES: {
+                COLLECTION_NAME: {"$ref": "#/definitions/collectionName"},
+                COLLECTION_PARAMETERS: {"$ref": "#/definitions/collectionParameters"},
+            },
+        },
+        "hookParameters": {
+            TYPE: OBJECT,
+            ADDITIONAL_PROPERTIES: False,
+            PATTERN_PROPERTIES: {
+                r"(.*){1,256}": {
+                    TYPE: "string",
+                    "pattern": r".*",
+                    "maxLength": 256,
+                }
+            },
+            "maxProperties": 20,
+        },
```

Comment:
maybe could be merged with the collectionParameters since they have the same schema

Reply:
Yes we can do that. I may have to change the name to collectionOrHookParameters in that case?